Kconfig.zephyr

```diff
@@ -623,6 +623,14 @@ config MCUBOOT_EXTRA_IMGTOOL_ARGS
 	  you can use this option to pass extra options to imgtool.
 	  For example, you could set this to "--version 1.2".
 
+config MCUBOOT_FORCE_IMAGE_GENERATION
```

Comment:
I'd really like to discourage accidentally creating unsigned images, and would prefer the build to fail if the key is not set. If someone wants to use mcuboot without signed images, they should have to explicitly request it. Perhaps another config that allows that.

Reply:
@d3zd3z @tejlmand So the current approach of adding a config option to generate unsigned images should work, right? Its default value is set to `n`, so it will not accidentally generate unsigned images.

If it looks good, I'll just rename `MCUBOOT_FORCE_IMAGE_GENERATION` to `MCUBOOT_UNSIGNED_IMAGE` as per @tejlmand's suggestion.

Reply:
> I'd really like to discourage accidentally creating unsigned images, and would prefer the build to fail if the key is not set.

Thanks for commenting.
Let's keep a dedicated setting for this.